node/src/accountant/scanners_tools.rs

```diff
@@ -0,0 +1,539 @@
+        fingerprint: &PendingPayableFingerprint,
+        max_pending_interval: u64,
+        logger: &Logger,
+    ) -> PendingTransactionStatus {
+        info!(
+            logger,
+            "Pending transaction '{}' couldn't be confirmed at attempt \
+            {} at {}ms after its sending",
+            fingerprint.hash,
+            fingerprint.attempt_opt.expectv("initialized attempt"),
+            elapsed_in_ms(fingerprint.timestamp)
+        );
+        let elapsed = fingerprint
+            .timestamp
+            .elapsed()
+            .expect("we should be older now");
+        let transaction_id = PendingPayableId {
+            hash: fingerprint.hash,
+            rowid: fingerprint.rowid_opt.expectv("initialized rowid"),
+        };
+        if max_pending_interval <= elapsed.as_secs() {
+            error!(
+                logger,
+                "Pending transaction '{}' has exceeded the maximum pending time \
+                ({}sec) and the confirmation process is going to be aborted now \
+                at the final attempt {}; manual resolution is required from the \
+                user to complete the transaction.",
+                fingerprint.hash,
+                max_pending_interval,
+                fingerprint.attempt_opt.expectv("initialized attempt")
+            );
+            PendingTransactionStatus::Failure(transaction_id)
+        } else {
+            PendingTransactionStatus::StillPending(transaction_id)
+        }
+    }
+
+    pub fn handle_status_with_success(
+        fingerprint: &PendingPayableFingerprint,
+        logger: &Logger,
+    ) -> PendingTransactionStatus {
+        info!(
+            logger,
+            "Transaction '{}' has been added to the blockchain; detected locally at attempt \
+            {} at {}ms after its sending",
+            fingerprint.hash,
+            fingerprint.attempt_opt.expectv("initialized attempt"),
+            elapsed_in_ms(fingerprint.timestamp)
+        );
+        PendingTransactionStatus::Confirmed(fingerprint.clone())
+    }
+
+    pub fn handle_status_with_failure(
+        fingerprint: &PendingPayableFingerprint,
+        logger: &Logger,
+    ) -> PendingTransactionStatus {
+        error!(
+            logger,
+            "Pending transaction '{}' announced as a failure, interpreting attempt \
+            {} after {}ms from the sending",
+            fingerprint.hash,
+            fingerprint.attempt_opt.expectv("initialized attempt"),
+            elapsed_in_ms(fingerprint.timestamp)
+        );
+        PendingTransactionStatus::Failure(fingerprint.into())
+    }
+}
+
+pub(crate) mod receivable_scanner_tools {
+    use crate::accountant::receivable_dao::ReceivableAccount;
+    use std::time::{Duration, SystemTime};
+
+    pub(crate) fn balance_and_age(
+        time: SystemTime,
+        account: &ReceivableAccount,
+    ) -> (String, Duration) {
+        let balance = format!("{}", (account.balance as f64) / 1_000_000_000.0);
```

Comment:
Please remove this division by one billion ( 10^9 ?).
We should show it more accurately than we're doing here. We can display Wei. No big deal.